File: Java/q1.java
import javax.swing.text.View;
import java.util.Random;

public class q1{
    private double x[][];

    public q1(double x[][]){
       this.x = x;
    }

    public void rowTest(){
        rowInit();
    }

    public void columnTest(){
        colInit();
    }

    private void rowInit(){
        int n = x.length;        
        Random r = new Random();
        long start = System.currentTimeMillis();       
        for(int i = 0;i<n;i++){
            for(int j = 0; j<n; j++){
                x[i][j] = r.nextDouble();
            }            
        }
        long finish = System.currentTimeMillis();       
        System.out.println(" Total time: " + (finish-start) + "ms.");
    }

    private void colInit(){
        int n = x.length;
        Random r = new Random();
        long start = System.currentTimeMillis();  
        for(int i = 0;i<n;i++){
            for(int j = 0; j<n; j++){
                x[j][i] = r.nextDouble();
            }            
        }
        long finish = System.currentTimeMillis();   
        System.out.println(" Total time: " + (finish-start) + "ms.");
    }


}

File: C#/q1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    public class q1
    {
        private double[,] x;

        public q1(double[,] x)
        {
            this.x = x;
        }

        public void rowTest()
        {
            rowInit();
        }

        public void colTest()
        {
            colInit();
        }

        private void rowInit()
        {
            int n = 20;
            Random r = new Random();
            var timer = System.Diagnostics.Stopwatch.StartNew();
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    x[i,j] = r.NextDouble();
                }
            }
            timer.Stop();
            var elapsedMs = timer.ElapsedMilliseconds;
            Console.WriteLine(elapsedMs);
        }

        private void colInit()
        {
            int n = 20;
            Random r = new Random();
            var timer = System.Diagnostics.Stopwatch.StartNew();
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < n; j++)
                {
                    x[j,i] = r.NextDouble();
                }
            }
            timer.Stop();
            var elapsedMs = timer.ElapsedMilliseconds;
            Console.WriteLine(elapsedMs);
        }


    }
}
